components/integrations/bots/bots.jsx

```diff
@@ -0,0 +1,206 @@
+
+            /**
+            * Load owner of bot account
+            */
+            getUser: PropTypes.func.isRequired,
+
+            /**
+            * Disable a bot
+            */
+            disableBot: PropTypes.func.isRequired,
+
+            /**
+            * Enable a bot
+            */
+            enableBot: PropTypes.func.isRequired,
+        }),
+
+        /**
+        *  Only used for routing since backstage is team based.
+        */
+        team: PropTypes.object.isRequired,
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            loading: true,
+        };
+    }
+
+    componentDidMount() {
+        this.props.actions.loadBots(
+            Constants.Integrations.START_PAGE_NUM,
+            Constants.Integrations.PAGE_SIZE
+        ).then(
+            (result) => {
+                if (result) {
+                    // We don't need to wait for this and we need to accept failure in the case where bot.owner_id is a plugin id
+                    result.data.map((bot) => this.props.actions.getUser(bot.owner_id));
+                    Promise.all(
+                        result.data.map((bot) => this.props.actions.getUserAccessTokensForUser(bot.user_id)).
+                            concat(result.data.map((bot) => this.props.actions.getUser(bot.user_id)))).
+                        then(() => {
+                            this.setState({loading: false});
+                        });
+                }
+            }
+        );
+    }
+
+    DisabledSection(props) {
+        if (!props.hasDisabled) {
+            return null;
+        }
+        const botsToDisplay = React.Children.map(props.disabledBots, (child) => {
+            return React.cloneElement(child, {filter: props.filter});
+        });
+        return (
+            <React.Fragment>
+                <div className='bot-disabled'>
+                    <FormattedMessage
+                        id='bots.disabled'
+                        defaultMessage='Disabled'
+                    />
+                </div>
+                <div className='bot-list__disabled'>
+                    {botsToDisplay}
+                </div>
+            </React.Fragment>
+        );
+    }
+
+    EnabledSection(props) {
+        const botsToDisplay = React.Children.map(props.enabledBots, (child) => {
+            return React.cloneElement(child, {filter: props.filter});
+        });
+        return (
+            <div>
+                {botsToDisplay}
+            </div>
+        );
+    }
+
+    render() {
+        const bots = Object.values(this.props.bots).sort((a, b) => a.username.localeCompare(b.username));
+        const botToJSX = (bot) => {
+            return (
+                <Bot
+                    key={bot.user_id}
+                    bot={bot}
+                    owner={this.props.owners[bot.user_id]}
+                    accessTokens={this.props.accessTokens[bot.user_id] || {}}
```

Comment:
The `|| {}` will cause the child component to re-render unnecessarily before the access token is loaded. Perhaps we could remove that and have the `Bot` component specify a value for that in `defaultProps`?

Reply:
Not sure what you mean. I can't specify a default for every possible key in the object.

Reply:
I mean that instead of
```javascript
<Bot accessTokens={accessTokens || {}}/>

class Bot {}
```
this could be
```
<Bot accessTokens={accessTokens}/>

class Bot {
    static defaultProps = {
        accessTokens: {},
    }
}
```
which wouldn't cause the underlying PureComponent to re-render whenever `Bots` does. Regardless, it's likely not much of a concern at this point since this only happens on load, and it's an inexpensive component anyway.